### PR TITLE
[WiP] Add middleware tests

### PR DESCRIPTION
**Pull request by wo0dyn**

### CODE REVIEW

File: oscar_impersonate/test_middleware.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                                                              response=self.response)
+        self.assertEqual(processed_response.content, CONTENT)
+
+    def test_streaming_response(self):
+        from django.http import StreamingHttpResponse
+        processed_response = self.middleware.process_response(request=self.request,
+                                                              response=StreamingHttpResponse())
+        self.assertIsInstance(processed_response, StreamingHttpResponse)
+
+    def test_gzip_content_encoding(self):
```
Comment (twidi) [inline oscar_impersonate/test_middleware.py:72]: don't see anything gzip related in the method

File: oscar_impersonate/test_middleware.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        self.response['Content-Type'] = 'application/json'
+        processed_response = self.middleware.process_response(request=self.request,
+                                                              response=self.response)
+        self.assertEqual(processed_response.content, CONTENT)
+        self.assertEqual(self.response.get('Content-Type'), 'application/json')
+
+    def test_non_html_content_type(self):
```
Comment (twidi) [inline oscar_impersonate/test_middleware.py:79]: hmm i think the methods name are inverted (with the above one)

File: oscar_impersonate/test_middleware.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        self.assertEqual(processed_response.content, CONTENT)
+        self.assertEqual(processed_response.get('Content-Encoding'), 'gzip')
+
+    def test_response_content_without_body_open_tag(self):
+        content_without_body_open_tag = b'Lorem lipsum'
+
+        self.response.content = content_without_body_open_tag
+        processed_response = self.middleware.process_response(request=self.request,
+                                                              response=self.response)
+        self.assertEqual(processed_response.content, content_without_body_open_tag)
```
Comment (twidi) [inline oscar_impersonate/test_middleware.py:92]: what is the result of this test with an impersonation ? Here there is none, so we really expect the result to be the same, but with one ?

File: tests/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Django framework
+django==1.6.5
```
Comment (twidi) [inline tests/requirements.txt:2]: ?

File: oscar_impersonate/test_middleware.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+</html>"""
+
+
+class OscarImpersonateMiddlewareTestCase(TestCase):
+
+    def setUp(self):
+        USERS = (
+            # username    is_superuser  is_staff
+            ('superuser', True,         True    ),  # noqa
+            ('staff',     True,         False   ),  # noqa
```
Comment (twidi) [inline oscar_impersonate/test_middleware.py:43]: this user is not used in test. Missing tests ?

File: tests/requirements.txt (hunk 4 above)
Comment (wo0dyn) [inline tests/requirements.txt:2]: Maybe we should use `tox` to test several versions of Django.
Here, was the latest version when I wrote these tests (a couple of months ago).

**Comment by wo0dyn** [comment]
This PR won't be merged. We need to refactor the whole app anyway, to avoid having a middleware here, when the developer only has to add short (and customizable) HTML snippet into their project.

**Pull request closed without merging**